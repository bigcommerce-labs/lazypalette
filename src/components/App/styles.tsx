import styled from 'styled-components';

export const StyledApp = styled.div`
    background: ${({ theme: { colors } }) => colors.background};
    align-content: flex-start;
    display: flex;
    flex-direction: column;
    height: 100%;
    position: fixed;
    width: 100%;
`;

export const Viewport = styled.div`
    display: flex;
    flex-direction: row;
    flex-grow: 1;
`;
