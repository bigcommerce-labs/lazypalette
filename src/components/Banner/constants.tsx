import React from 'react';

export const CopyMessage = 'Link copied to clipboard.';

export const TooltipMessage =
    <>
        <p>You must launch your store from the Getting Started page in order to make it public.</p>
        <p><a href="/manage/getting-started">Go to Getting Started</a></p>
        <p>You can still let other people see your store before it is launched by sharing the private store link.</p>
    </>;

export const MaintenanceMessage = 'Your store is down for maintenance.';
export const PrelaunchMessage = 'Your store has not launched and is not publicly visible.';
