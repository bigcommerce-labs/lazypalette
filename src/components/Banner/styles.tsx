import styled from 'styled-components';

export const StyledBanner = styled.div`
    background-color: #FFD24D;
    text-align: center;
    width: 100%;
    line-height: 2.5rem;
    font-size: 1rem;
    font-weight: 600;
    color: #303540;
`;

export const StyledIcon = styled.span`
    span {
        margin-bottom: -5px;
    }
`;

export const StyledMessage = styled.p`
    text-align: left;

    a {
        color: ${({ theme }) => theme.colors.primary};
        display: block;
        font-weight: bold;
        margin-top: 5px;
        text-decoration: none;
    }
`;

export const StyledShareableLink = styled.button`
    background: ${({ theme }) => theme.colors.warning};
    color: ${({ theme }) => theme.colors.primaryText};
    font-size: ${({ theme }) => theme.typography.fontSize.small};
    font-weight: ${({ theme }) => theme.typography.fontWeight.bold}
    font-family: ${({ theme }) => theme.typography.fontFamily.headings};
    margin: .5rem 0 .5rem 1.25rem;
    outline: none;
    border: 1px solid;
    border-radius: 2px;
    display: inline-block;
    padding: .25rem .5rem;
    line-height: 1.375rem;
    cursor: pointer;
`;
