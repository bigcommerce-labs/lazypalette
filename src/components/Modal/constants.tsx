export const enum Messages {
    AlertTitle = 'Alert',
    Cancel = 'Cancel',
    ConfirmTitle = 'Please confirm.',
    Ok = 'Ok',
    Reset = 'This theme has unpublished changes. Do you want to proceed?',
    Variation = 'This variation has unpublished changes. Do you want to proceed?',
}
