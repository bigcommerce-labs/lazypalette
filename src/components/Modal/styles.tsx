import styled from 'styled-components';

interface OverlayProps {
    isTransparent?: boolean;
}

export const Close = styled.div`
    cursor: pointer;
    text-align: center;
    width: 2rem;
    height: 2rem;
    margin-right: .25rem;
    display: flex;
    justify-content: center;
    align-items: center;

    :hover {
      cursor: pointer;
      background: ${({ theme }) => theme.colors.darkHover};
      border-radius: 50%;
    }
`;

Close.defaultProps = {
    theme: {
        colors: {
            darkHover: '#D3D9E5',
            primaryText: '#AAAAAA',
        },
    },
};

export const Container = styled.div`
    position: fixed;
    top: 0;
    left: 0;
    height: 100%;
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
    overflow-y: auto;
    z-index: ${({ theme }) => theme.layers.higher};
`;

Container.defaultProps = {
    theme: {
        layers: {
            higher: 100,
        },
    },
};

export const Overlay = styled.div.attrs<OverlayProps>({})`
    position: fixed;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    overflow-y: auto;
    opacity: ${({ isTransparent }) => isTransparent ? 0 : 0.45};
    background: #000000;
    z-index: ${({ theme }) => theme.layers.higher};
`;

Overlay.defaultProps = {
    theme: {
        layers: {
            higher: 100,
        },
    },
};

export const Content = styled.div`
    color: ${({ theme }) => theme.colors.primaryText};
    line-height: 1.5rem;
    font-size: ${({ theme }) => theme.typography.fontSize.small};
    font-weight: ${({ theme }) => theme.typography.fontWeight.normal};
    margin: 0;
    padding: 0;
`;

Content.defaultProps = {
    theme: {
        colors: {
            primaryText: '#AAAAAA',
        },
        typography: {
            fontSize: {
                small: '16px',
            },
            fontWeight: {
                normal: 400,
            },
        },
    },
};

export const Header = styled.div`
    align-items: stretch;
    display: flex;
    flex-shrink: 0;
    justify-content: space-between;
    line-height: 2rem;
    margin: 0 -0.75rem 1rem 0;
    padding: 0;
`;

export const Title = styled.h3`
    color: ${({ theme }) => theme.colors.primaryText};
    font-size: ${({ theme }) => theme.typography.fontSize.larger};
    font-weight: ${({ theme }) => theme.typography.fontWeight.normal};
    margin: 0;
`;

Title.defaultProps = {
    theme: {
        colors: {
            primaryText: '#303540',
        },
        typography: {
            fontSize: {
                larger: '24px',
            },
            fontWeight: {
                normal: 600,
            },
        },
    },
};

export const ModalBox = styled.div`
    background-color: ${({ theme }) => theme.colors.empty};
    border: 1px solid ${({ theme }) => theme.colors.guideText};
    border-radius: 4px;
    box-shadow: ${({ theme }) => theme.elevation.floating};
    box-sizing: border-box;
    overflow: auto;
    padding: 1.5rem;
    position: relative;
    width: 30rem;
    z-index: ${({ theme }) => theme.layers.highest};
`;

ModalBox.defaultProps = {
    theme: {
        colors: {
            empty: '#FFFFFF',
            guideText: '#ABB1BE',
        },
        elevation: {
            floating: '0 2px 12px rgba(48, 53, 64, 0.2)',
        },
        layers: {
            highest: 500,
        },
    },
};

export const ModalBody = styled.div`
`;

export const ConfirmButtons = styled.div`
    display: inline-flex;
    justify-content: flex-end;
`;

export const ConfirmButton = styled.div`
    margin-left: 1rem;
`;

export const ModalView = styled.div`
    max-width: 35rem;
`;

export const ModalFooter = styled.div`
    display: flex;
    justify-content: flex-end;
    margin: 1.5rem 0 0 0;
`;
