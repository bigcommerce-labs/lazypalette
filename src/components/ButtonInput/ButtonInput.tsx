import React, { MouseEvent, MouseEventHandler, PureComponent } from 'react';

import { Button, Wrapper } from './styles';

interface ButtonInputProps extends Partial<{
    border: boolean;
    children: JSX.Element | string;
    classType: string;
    disabled: boolean;
    onClick: MouseEventHandler<HTMLButtonElement>;
    testId?: string;
    type: string;
}> {}

class ButtonInput extends PureComponent<ButtonInputProps> {
    handleClick = (e: MouseEvent<HTMLButtonElement>) => {
        if (this.props.onClick) {
            this.props.onClick(e);
        }
    };

    render() {
        const {
            border = true,
            disabled,
            classType,
            testId,
            type,
            children,
        } = this.props;

        return (
            <Wrapper>
                <Button
                    border={border}
                    disabled={disabled}
                    onClick={this.handleClick}
                    classType={classType}
                    type={type}
                    {...(testId ? {'data-test-id': testId} : {})}
                >
                    {children}
                </Button>
            </Wrapper>
        );
    }
}

export default ButtonInput;
