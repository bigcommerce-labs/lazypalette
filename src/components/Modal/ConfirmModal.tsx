import React, { PureComponent } from 'react';

import ButtonInput from '../ButtonInput/ButtonInput';

import { Confirm, ConfirmBody, ConfirmButtons, ConfirmFooter } from './styles';
import Modal from './Modal';

interface ConfirmModalProps extends Partial<{
    body: string;
    title: string;
    primaryAction(): void;
    secondaryAction(): void;
    overlayClose(): void;
}> {}

const CTAButtons = ({ primaryAction, secondaryAction }: ConfirmModalProps) => (
    <ConfirmButtons>
        <ButtonInput
            onClick={primaryAction}
            classType="primary"
            type="button"
        >
            Cancel
        </ButtonInput>
        <ButtonInput
            onClick={secondaryAction}
            type="button"
        >
            OK
        </ButtonInput>
    </ConfirmButtons>
);

class ConfirmModal extends PureComponent<ConfirmModalProps> {
    static defaultProps: ConfirmModalProps = {
        title: 'Please Confirm:',
    };

    render() {
        const {
            body,
            primaryAction,
            secondaryAction,
            overlayClose,
            title,
        } = this.props;

        return (
            <Modal
                onClose={primaryAction}
                overlayClose={overlayClose}
                title={title}
                isTransparent={false}
            >
                <Confirm>
                    {body &&
                        <ConfirmBody>{body}</ConfirmBody>}
                    <ConfirmFooter>
                        <CTAButtons
                            primaryAction={primaryAction}
                            secondaryAction={secondaryAction}
                        />
                    </ConfirmFooter>
                </Confirm>
            </Modal>
        );
    }
}

export default ConfirmModal;
