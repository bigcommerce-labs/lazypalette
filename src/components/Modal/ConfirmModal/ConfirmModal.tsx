import React, { PureComponent } from 'react';

import ButtonInput from '../../ButtonInput/ButtonInput';

import { ConfirmButtons, ModalBody, ModalFooter, ModalView } from '../styles';
import Modal from '../Modal';

import { Messages } from '../constants';

interface ConfirmModalProps extends Partial<{
    body: string;
    title: string;
    primaryAction(): void;
    secondaryAction(): void;
    overlayClose(): void;
}> {}

const CTAButtons = ({ primaryAction, secondaryAction }: ConfirmModalProps) => (
    <ConfirmButtons>
        <ButtonInput
            onClick={primaryAction}
            classType="primary"
            type="button"
        >
            {Messages.Cancel}
        </ButtonInput>
        <ButtonInput
            onClick={secondaryAction}
            type="button"
        >
            {Messages.Ok}
        </ButtonInput>
    </ConfirmButtons>
);

class ConfirmModal extends PureComponent<ConfirmModalProps> {
    static defaultProps: ConfirmModalProps = {
        title: Messages.ConfirmTitle,
    };

    render() {
        const {
            body,
            primaryAction,
            secondaryAction,
            overlayClose,
            title,
        } = this.props;

        return (
            <Modal
                onClose={primaryAction}
                overlayClose={overlayClose}
                title={title}
                isTransparent={false}
            >
                <ModalView>
                    {body &&
                        <ModalBody>{body}</ModalBody>}
                    <ModalFooter>
                        <CTAButtons
                            primaryAction={primaryAction}
                            secondaryAction={secondaryAction}
                        />
                    </ModalFooter>
                </ModalView>
            </Modal>
        );
    }
}

export default ConfirmModal;
