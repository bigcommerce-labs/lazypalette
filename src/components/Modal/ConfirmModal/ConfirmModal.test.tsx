import { mount, shallow } from 'enzyme';
import React from 'react';

import ButtonInput from '../../ButtonInput/ButtonInput';

import { Messages } from '../constants';
import { ModalBody } from '../styles';

import ConfirmModal from './ConfirmModal';

describe('ConfirmModal', () => {
    const mockHandler = jest.fn();
    const mockBody = 'This theme will self-destruct in T-minus';
    const mockTitle = 'Please Confirm:';

    it('renders', () => {
        const confirmModal = shallow(
            <ConfirmModal
                title={mockTitle}
                primaryAction={mockHandler}
                secondaryAction={mockHandler}
            />
        );

        expect(confirmModal).toMatchSnapshot();
    });

    describe('children prop', () => {
        it('should render ConfirmBody when children passed', () => {
            const modal = shallow(
                <ConfirmModal
                    title={mockTitle}
                    primaryAction={mockHandler}
                    secondaryAction={mockHandler}
                >
                    {mockBody}
                </ConfirmModal>
            );

            expect(modal.find(ModalBody).prop('children')).toEqual(mockBody);
        });
    });

    describe('cancel button', () => {
        describe('when the user clicks the cancel button', () => {
            it('should call the handler', () => {
                const mockSecondaryAction = jest.fn();

                const modal = mount(
                    <ConfirmModal
                        title={mockTitle}
                        primaryAction={mockHandler}
                        secondaryAction={mockSecondaryAction}
                    />
                );

                const cancelButton = modal.find(ButtonInput).find({ children: Messages.Cancel }).last();
                cancelButton.simulate('click');
                expect(mockSecondaryAction).toHaveBeenCalledTimes(1);
            });
        });
    });

    describe('ok button', () => {
        describe('when the user clicks the ok button', () => {
            it('should call the handler', () => {
                const mockPrimaryAction = jest.fn();

                const modal = mount(
                    <ConfirmModal
                        title={mockTitle}
                        primaryAction={mockPrimaryAction}
                        secondaryAction={mockHandler}
                    />
                );

                const okButton = modal.find(ButtonInput).find({ children: Messages.Ok }).last();
                okButton.simulate('click');
                expect(mockPrimaryAction).toHaveBeenCalledTimes(1);
            });
        });
    });
});
