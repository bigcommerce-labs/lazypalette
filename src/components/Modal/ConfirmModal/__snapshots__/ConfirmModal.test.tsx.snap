// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmModal renders 1`] = `
<Modal
  isTransparent={false}
  onClose={[MockFunction]}
  title="Please Confirm:"
>
  <styled.div>
    <styled.div>
      This theme will self-destruct in T-minus
    </styled.div>
    <styled.div>
      <CTAButtons
        primaryAction={[MockFunction]}
        secondaryAction={[MockFunction]}
      />
    </styled.div>
  </styled.div>
</Modal>
`;
